### DISCUSSION TITLE
Functions don't show documentation in autocomplete

### DISCUSSION THREAD

**Discussion by robbins** [mfussenegger/nvim-jdtls#601] [Q&A]
When in the suggestions list from `nvim-cmp` using nvim-lspconfig, the function signatures show up, but no documentation.
![image](https://github.com/mfussenegger/nvim-jdtls/assets/31457698/b83e4a2f-2dc4-4659-b43f-5cd2b754a5cc)

The LSPs go-to-definition feature works, and the JavaDocs are present for classes and functions.
![image](https://github.com/mfussenegger/nvim-jdtls/assets/31457698/33790b28-47b0-43ea-ab44-64f747aad64f)

I have enabled signatureHelp:
```
  settings = {
    java = {
      signatureHelp = { enabled = true },
    }
  },
```

I found this related discussion https://github.com/mfussenegger/nvim-jdtls/discussions/124, but I am on NixOS and don't have a `jdk-*-sources` package. However, they seem to be included in my currently installed JDK as I am able to go to definition.

![image](https://github.com/mfussenegger/nvim-jdtls/assets/31457698/48fbc0d7-54a5-4d13-b2a2-347c71f8e9d2)
Documentation is available for other languages, like C, so my LSP config appears to be correct.

**Comment by B1kku** [comment]
I had this same issue, signatureHelp had nothing to do with it, this was due to cmp capabilities not being added to the lsp.
```lua
config = {
   ...,
   capabilities = require("cmp_nvim_lsp").default_capabilities(),
   ...,
}
```
I'm surprised this option isn't listed on the README and completion isn't mentioned either given how important it is.

**Reply by brandishcode** [reply]
Omg. I have been searching for this! Thank my good sir. Yeah, kind of weird it is not included in README.
